### DISCUSSION TITLE
Getting a list of healthy/unhealthy clusters

### DISCUSSION THREAD

**Discussion by PieterBoeren** [dotnet/yarp#1124] [Q&A] [answered]
Is it possible to get a list of healthy and unhealthy clusters? I'm adding health checks to my clusters, but I would love to have some way to get which clusters are healthy (or even better, how many destinations of that cluster are healthy) and which not.

I basically want to build a dashboard to show the current status of the API's, but as far as I can tell, there is no easy way to get the status of the clusters.

**Comment by Tratcher** [comment, marked as answer]
We've explored similar ideas (https://github.com/microsoft/reverse-proxy/issues/123) but haven't tried building this out yet. In theory you can build this today. Note clusters are not marked healthy or unhealthy, only destinations have heath state.
- Implement [IClusterChangeListener](https://github.com/microsoft/reverse-proxy/blob/1f27ecf4e1b23fd44bc9106f6765c65944caa271/src/ReverseProxy/Model/IClusterChangeListener.cs#L15) and add it as a singleton in the DI container. This will let you collect the cluster instances as they're added to the system. 
- In your health dashboard you check each cluster
  - Check each destination in [ClusterState.Destinations](https://github.com/microsoft/reverse-proxy/blob/1f27ecf4e1b23fd44bc9106f6765c65944caa271/src/ReverseProxy/Model/ClusterState.cs#L46).
    - Check DestinationState.[Health](https://github.com/microsoft/reverse-proxy/blob/1f27ecf4e1b23fd44bc9106f6765c65944caa271/src/ReverseProxy/Model/DestinationState.cs#L48).

Let us know how that works for you.

**Reply by PieterBoeren** [reply]
Yes, this does work. An event or something would be easier, but this will suffice. Thanks for the pointers.

Discovered a bug though: #1134
